src/t_list.c

```diff
@@ -33,63 +33,51 @@
  * List API
  *----------------------------------------------------------------------------*/
 
-void listTypeConvertToQuicklist(robj *o) {
-    serverAssert(o->encoding == OBJ_ENCODING_LISTPACK);
-
-    quicklist *ql = quicklistCreate();
-    quicklistSetOptions(ql, server.list_max_listpack_size, server.list_compress_depth);
-
-    /* Append listpack to quicklist if it's not empty, otherwise release it. */
-    if (lpLength(o->ptr))
-        quicklistAppendListpack(ql, o->ptr);
-    else
-        lpFree(o->ptr);
-    o->ptr = ql;
-    o->encoding = OBJ_ENCODING_QUICKLIST;
-}
-
-void listTypeConvertToListpack(robj *o) {
-    quicklist *ql = o->ptr;
-    serverAssert(o->encoding == OBJ_ENCODING_QUICKLIST && ql->len == 1 &&
-        ql->head->container == QUICKLIST_NODE_CONTAINER_PACKED);
-
-    /* Extract the listpack from the unique quicklist node,
-     * then reset it and release the quicklist. */
-    o->ptr = ql->head->entry;
-    ql->head->entry = NULL;
-    quicklistRelease(ql);
-    o->encoding = OBJ_ENCODING_LISTPACK;
-}
-
-/* Check the length and size of a number of objects that will be added to list to see
- * if we need to convert a listpack to a quicklist. Note that we only check string
- * encoded objects as their string length can be queried in constant time.
+/* Check the length and size of a listpack to see if we need to convert it to quicklist.
+ *
+ * If 'growing' is 1, we also check the length and size of a number of objects that will
+ * be added to list to see if we need to convert a listpack to a quicklist. Note that we
+ * only check string encoded objects as their string length can be queried in constant time.
  *
  * If callback is given the function is called in order for caller to do some work
  * before the list conversion. */
-void listTypeTryConversionForGrowing(robj *o, robj **argv, int start, int end,
-                                     beforeConvertCB fn, void *data)
+static void listTypeTryConvertListpack(robj *o, int growing,
+                                       robj **argv, int start, int end,
+                                       beforeConvertCB fn, void *data)
 {
     if (o->encoding != OBJ_ENCODING_LISTPACK) return;
 
     size_t sz_limit;
     unsigned long count_limit;
-    size_t sum = 0;
-
-    for (int i = start; i <= end; i++) {
-        if (!sdsEncodedObject(argv[i]))
-            continue;
-        sum += sdslen(argv[i]->ptr);
+    size_t add_bytes = 0;
+    size_t add_length = 0;
+
+    if (growing) {
```

Comment:
it doesn't look like this function needs that `growing` argument, it can just check that argv is non-null

Reply:
Done